dvc/remote/base.py

```diff
@@ -285,10 +286,13 @@ def load_dir_cache(self, checksum):
             )
             return []
 
-        for info in d:
-            # NOTE: here is a BUG, see comment to .as_posix() below
-            relative_path = PathInfo.from_posix(info[self.PARAM_RELPATH])
-            info[self.PARAM_RELPATH] = relative_path.fspath
+        if os.name == "nt":
+            # only need to convert it for Windows
```

Comment:
Actually this is wrong, this code is not only for local paths. Looking into it...